doc/sphinx/src/models.rst

```diff
@@ -3,3 +3,409 @@
+energy, free energy, and entropy consistent with the constraints of equillibrium
+thermodynamics.
+
+``singularity-eos`` contains a number of equations of state that are most useful
+in hydrodynamics codes. All of the equations of state presented here are
+considered "complete" equations of state in the sense that they contain all the
+information needed to derive a given thermodynamic quantity. However, not all
+variables are exposed since the emphasis is on only those that are needed for
+hydrodynamics codes. An incomplete equation of state is often sufficient for
+pure hydrodynamics since it can relate pressure to a given energy-density state,
+but it is missing information that relates the energy to temperature (which in
+turn provides a means to calculate entropy).
+
+The Mie-Gruneisen form
+````````````````````````
+
+Many of the following equations of state are considered to be of
+the "Mie-Gruneisen form", which has many implications but for our purposes
+means that the general form of the EOS is
+
+.. math::
+
+    P - P_\mathrm{ref} = \rho \Gamma(\rho) (E - E_\mathrm{ref})
+
+where 'ref' denotes quantities along some reference curve, :math:`P` is the
+pressure, :math:`\rho` is the density, :math:`\Gamma` is the Gruneisen
+parameter, and :math:`E` is the specific internal energy. In this sense, an EOS
+of this form uses the Gruneisen parameter to describe the pressure behavior of
+the EOS away from the reference curve. Coupled with a relationship between
+energy and temperature (sometimes as simple as a constant heat capacity), the
+complete equation of state can be constructed.
+
+To some degree it is the complexity of the reference state and the heat
+capacity that will determine an EOS's ability to capture the complex behavior of
+a material. At the simplest level, the ideal gas EOS uses a reference state at
+zero pressure and energy, while more complex equations of state such as the
+Davis EOS use the material's isentrope. In ths way, the reference curve
+indicates the conditions under which you can expect the EOS to represent the
+intended behavior.
+
+Nomenclature
+---------------------
+
+The EOS models in ``singularity-eos`` are defined for the following sets of
+dependent and independent variables through various member functions described
+in the :doc:`EOS API <using-eos>`.
+
++--------------------------+----------------------+------------------------+
+| Function                 | Dependent Variable   | Independent Variables  |
++==========================+======================+========================+
+| :math:`T(\rho, E)`       | Temperature          | Density, Energy        |
++--------------------------+----------------------+                        |
+| :math:`P(\rho, E)`       | Pressure             |                        |
++--------------------------+----------------------+------------------------+
+| :math:`E(\rho, T)`       | Energy               | Density, Temperature   |
```

Comment:
`singularity-eos` never uses total energy. Basically everywhere it uses specific internal energy sie, or \varepsilon.

Reply:
I'll change this to internal energy to differentiate.